classes/Model/AnnouncementManagement.php

```diff
@@ -18,7 +18,7 @@ function is_valid_item($item)
     if (isset($item->lang)) {
         $language = $item->lang;
     } else {
-        $language = "en";
+        $language = $current_language;
```

Comment:
Think this is the wrong "en" to change, I'm afraid - defaulting to English if not present in the item is fine, but it's the check at line 46 that should check $language against $current_language, not "en".

Reply:
Doh, yes.